pygribjump/src/pygribjump/pygribjump.py

```diff
@@ -317,6 +345,31 @@ def extract_single(self, request : dict[str, str | list], ranges : list[tuple[in
         logctx_c = ffi.new('const char[]', logctx.encode('ascii'))
         return ExtractionSingleIterator(self.ctype, request, ranges, gridHash, logctx_c)
 
+    # Convenience functions for extracting from masks and indices
+    def extract_from_mask(self, requests : list[dict[str, str]], masks : np.ndarray, gridHash: str = None, ctx=None) -> ExtractionIterator:
+        
+        if type(requests) is not list:
```

Comment:
[nit] What would you think about changing this to the following more relaxed check so we don't raise an error if someone (for whatever reason) passes something that's a subclass of list?
```suggestion
        if not isinstance(requests, list):
```
Definitely nothing important though.

Reply:
Makes sense, this is what I do in the original extract method anwyay so it makes sense to be consistent.